src/Domain/JacobianDiagnostic.hpp

```diff
@@ -69,31 +106,20 @@ struct JacobianDiagnostic : db::SimpleTag {
 /// coordinates held by `MappedCoordsTag`. The coordinates must be in the target
 /// frame of the map. See `domain::jacobian_diagnostic` for details of the
 /// calculation.
-template <typename MapTag, typename SourceCoordsTag, typename MappedCoordsTag>
-struct JacobianDiagnosticCompute : JacobianDiagnostic, db::ComputeTag {
-  using base = JacobianDiagnostic;
+template <size_t Dim, typename TargetFrame, typename MappedCoordsTag>
+struct JacobianDiagnosticCompute : JacobianDiagnostic<Dim>, db::ComputeTag {
+  using base = JacobianDiagnostic<Dim>;
   using return_type = typename base::type;
-  using argument_tags =
-      tmpl::list<MapTag, SourceCoordsTag, MappedCoordsTag, Mesh<MapTag::dim>>;
-  static constexpr auto function(
-      const gsl::not_null<return_type*> jacobian_diag,
-      const typename MapTag::type& element_map,
-      const tnsr::I<DataVector, MapTag::dim, typename MapTag::source_frame>&
-          source_coords,
-      const tnsr::I<DataVector, MapTag::dim, typename MapTag::target_frame>&
-          mapped_coords,
-      const ::Mesh<MapTag::dim>& mesh) {
-    const auto analytic_jacobian =
-        determinant_and_inverse(element_map.inv_jacobian(source_coords)).second;
-    // Note: Jacobian has the source frame index second, but
-    // logical_partial_derivative prepends the logical (source frame) index.
-    // So this is actually the transpose of the numerical jacobian.
-    const auto numerical_jacobian_transpose =
-        logical_partial_derivative(mapped_coords, mesh);
-
-    ::domain::jacobian_diagnostic(jacobian_diag, analytic_jacobian,
-                                  numerical_jacobian_transpose);
-  }
+  using argument_tags = tmpl::list<
+      ::domain::Tags::InverseJacobian<Dim, Frame::ElementLogical, TargetFrame>,
+      MappedCoordsTag, Mesh<Dim>>;
```

Comment:
I think you can delete `MappedCoordsTag` and instead write `domain::Tags::Coordinates<Dim, TargetFrame>`.

Please update the docs of struct struct as well.